src/stream/quic_lb/ngx_stream_quic_lb_module.c

```diff
@@ -303,7 +303,7 @@ ngx_stream_quic_lb_downstream_pkt_send_process(void *s_,
         rrp->quic_lb_conf = s->quic_lb_conf;
 
         /* for initial pkt, do retry service, draft 04, chapter 6. */
-        if (rrp->quic_lb_conf->valid && rrp->pkt->initial_pkt) {
+        if (rrp->quic_lb_conf->valid && s->pkt->initial_pkt) {
```

Comment:
before do retry service, following needed check:

1. "The service SHOULD drop
packets with unknown key sequence, or an AEAD checksum that does not
match the expected value. (By construction, the AEAD checksum will
only match if the client IP Address also matches.)"


2. "The service SHOULD drop a packet containing a token where the ODCIL
is greater than zero and less than the minimum number of octets for a
client-generated CID (8 in QUIC version 1). The service also SHOULD
drop a packet containing a token where the ODCIL is zero and RSCIL is
nonzero."

Reply:
The full check is in `ngx_stream_quic_lb_validate_share_state_token`.
Another problem, draft demand that retry service should behind the NAT device, may be we should try forward packet first, then do retry service